class/trim_decorator.rb

```diff
@@ -0,0 +1,8 @@
+require_relative 'base_decorator'
+
+class TrimDecorator < Decorator
+  def correct_name
+    temp_name = @nameable.correct_name[0..9]
+    @nameable.correct_name.length >= 10 ? temp_name : @nameable.correct_name
+  end
```

Comment:
- [ ] [Optional] Well! you did a great job. your code looks good👍. but there is a lot of ways to do the same thing in a different way. I noticed in your correct_name method, you're using some conditional logic that can be simplified. You're essentially returning the first 10 characters of @nameable.correct_name, and you can achieve the same result with a more concise and readable code by directly slicing the string. 

```suggestion
  def correct_name
    @nameable.correct_name[0..9]
  end
```

Reply:
I'll implement the suggestion